espflash/src/connection/mod.rs

```diff
@@ -277,6 +281,48 @@ impl Connection {
                 info!("Staying in flasher stub");
                 Ok(())
             }
+            ResetAfterOperation::WatchdogReset => {
+                info!("Resetting device with watchdog");
+
+                match chip {
+                    Chip::Esp32c3 => {
+                        if pid == USB_SERIAL_JTAG_PID {
+                            wdt_reset(chip, self)?;
+                        }
+                    }
+                    Chip::Esp32p4 => {
+                        let esp32p4 = esp32p4::Esp32p4;
+                        // Check if the connection is USB OTG
+                        if esp32p4.connection_is_usb_otg(self)? {
+                            wdt_reset(chip, self)?;
+                        }
+                    }
+                    Chip::Esp32s2 => {
+                        let esp32s2 = esp32s2::Esp32s2;
+                        // Check if the connection is USB OTG
+                        if esp32s2.connection_is_usb_otg(self)? {
+                            // Check the strapping register to see if we can perform RTC WDT
+                            // reset
+                            if esp32s2.can_wtd_reset(self)? {
+                                wdt_reset(chip, self)?;
+                            }
+                        }
+                    }
+                    Chip::Esp32s3 => {
+                        let esp32s3 = esp32s3::Esp32s3;
+                        if pid == USB_SERIAL_JTAG_PID || esp32s3.connection_is_usb_otg(self)? {
+                            // Check the strapping register to see if we can perform RTC WDT
+                            // reset
+                            if esp32s3.can_wtd_reset(self)? {
+                                wdt_reset(chip, self)?;
+                            }
+                        }
+                    }
+                    _ => println!("Unsupported chip for watchdog reset {:?}", chip),
```

Comment:
This should probably be an error instead of a `println!`

Reply:
True